src/cursor/cur_hs.c

```diff
@@ -424,7 +444,8 @@ __curhs_prev_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
         WT_ERR(file_cursor->get_key(file_cursor, &btree_id, datastore_key, &start_ts, &counter));
 
         /* Stop before crossing over to the next btree. */
-        if (F_ISSET(hs_cursor, WT_HS_CUR_BTREE_ID_SET) && btree_id != hs_cursor->btree_id) {
+        if (F_ISSET(hs_cursor, WT_HS_CUR_BTREE_ID_SET) &&
+          !F_ISSET(std_cursor, WT_CURSTD_HS_READ_ACROSS_BTREE) && btree_id != hs_cursor->btree_id) {
```

Comment:
Do we really need this flag?

Reply:
We need to place the cursor at the end of the btree range. Currently what I do is to place the cursor at the first record that is larger than that btree and do a prev. To put the cursor on the first record of the next btree, I need to search with btree + 1, if without that flag, any cursor movement will result to not found, which is not I want.

Reply:
We need to write this information as a comment, later we forget the need for that flag.